src/Functions/array/arrayRemove.cpp

```diff
@@ -0,0 +1,65 @@
+#include <Functions/array/arrayRemove.h>
+#include <Functions/FunctionFactory.h>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int ILLEGAL_COLUMN;
+}
+
+ColumnPtr ArrayRemoveImpl::execute(const ColumnArray & array, ColumnPtr mapped)
```

Comment:
I don't see how the normal mode of operation:

```
arrayRemove(array, element)
```

could work.

Reply:
There are 3 things that can be done for this:

1. Overload the function arrayRemove.
2. add another argument in the same function as size_t element to remove ( or we could do index to remove from - easier complexity wise ).
3. add another function called arrayRemoveElememt: which does -> `arrayRemoveElement(arr,ele)`